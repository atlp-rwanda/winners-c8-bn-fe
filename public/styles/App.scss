h1,
h2 {
  color: darkred;
}

body {
  height: 100vh;
  background-color: #f5f5f5;
}

#root {
  height: 100%;
}

:root {
  font-size: 62.5% !important;
}
