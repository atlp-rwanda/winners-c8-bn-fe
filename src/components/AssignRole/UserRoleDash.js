import MenuItem from '@mui/material/MenuItem';
import React, { useEffect, useState } from 'react';
import "bootstrap/dist/css/bootstrap.min.css";
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import { Skeleton } from "@mui/material";
import Select from '@mui/material/Select';
import { fetchRoles, fetchUsers, assignRoles } from '../../redux/actions/roleActions';
import { useDispatch, useSelector } from 'react-redux';
import "./UserRoleDash.scss"
import Navbar from "../Navbar/Navbar";
import Sidebar from "../Sidebar/Sidebar";
const UserRoleDash = () => {
        const userInfos = useSelector((state) => state.users?.users?.data)
      
      
        const roleData = useSelector((state) => state.roles?.roles)
      
        useEffect(() => {
          }, [SelectedEmail])
      
      
        const [users, setUsers] = useState()
      
      
        const [isLoading, setIsLoading] = useState(false);
      
        const [SelectedRole, setSelectedRole]  = useState(null)
      
        const [SelectedEmail, setSelectedEmail]  = useState(null)
      
        const [roles, setRoles] = useState(null)
        
        const dispatch = useDispatch()
      
      
      useEffect(() => {
      fetchUsers()(dispatch)
      }, [])
      
      useEffect(() => {
        fetchRoles()(dispatch)
        }, [])
      
      
      
        const SubmitAssignRole = (event) =>{
        event.preventDefault();
        dispatch(assignRoles(SelectedEmail, SelectedRole))
        }
        
        useEffect(() => {
          setUsers({
            firstName:userInfos?.firstName,
            lastName:userInfos?.lastName,
            email:userInfos?.email,
          })
          setIsLoading(true)
        }, [userInfos])
      
          useEffect(() => {
            setRoles({
              id:roleData?.roleId,
              role:roleData?.roleName,
            })
            setIsLoading(true)
        }, [roleData])
    return ( 
        <>
      <div className="Content" data-testid='assign-role'>
      <Sidebar />
      <div className="ContentContainer">
        <Navbar />
        <div className="titleDashboard">
          <div className="text">
           <h4>Dashboard&gt;&gt;</h4>
           <h6>Assign Role</h6>
           </div>
           <h4>Dashboard&gt;&gt;</h4>
          </div>
        
  <div className='contentBlock'>
  <p>Assign Role</p>  
  {isLoading ? (<div className="Content">
<form className='formClass'>
<label className='labelClass'>Choose an Email</label>

<Select
 className='inputData'
 data-testid="user-byEmail"
defaultValue={users?.email}
  onChange={(event) =>
    setSelectedEmail(
       event.target.value,
    )
  }
 >
     {userInfos?.map(value => {
    return (
      <MenuItem value={value.email} key={value.email}>
        {value.email}
      </MenuItem>
    )
  })}
 </Select>
<div className='role'>Select Role name</div>
<Select
 className='OptionSelect'
 data-testid="role-name-byId"
  defaultValue={roles?.role}
  onChange={(event) =>
{setSelectedRole(event.target.value)}
  }
 >
   {roleData?.map(value =>{
     return ( <MenuItem  value={value.id} key={value.id}>
       {value.roleName}
     
     </MenuItem>)
   })}
 </Select>
<button
className='buttonClass'
type="submit"
data-testid="assign-role-btn"
onClick={SubmitAssignRole}

><div className='buttonText'>assignRole</div></button> 
<ToastContainer/>
</form>
</div>):
          <Skeleton
          variant="rectangle"
          animation="wave"
          width={1100}
          height={800}
          />}
</div>
      </div>
    </div>
    </>
     );
}
 
export default UserRoleDash;