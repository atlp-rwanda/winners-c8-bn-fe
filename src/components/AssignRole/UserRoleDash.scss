.Content {
    display: flex;
    background: #F5F5F5;
  
    .ContentContainer {
      flex: 6;
    }
    .titleDashboard{
        width: 95%;
        margin: 5px auto;
        display: flex;
        justify-content: space-between;
        align-items: center;

        .text{
          display: flex;
          align-items: center;
          padding: 20px;
          margin-top: 10px;
          gap: 20px;
          h3, h4,h6{
            color:#395B64;
            font-size: medium
            ;
          }
        }
    }
    .contentBlock{
      width: 95%;
      margin: 0px auto;
    }
    .labelClass{
        padding-left: 10px;
        position: absolute;
        font-style: normal;
        font-size: 15px;
        left: 220px;
        top: 250px;
        justify-content: space-between;
    }
    .inputData{
        position: absolute;
        width: 450px;
        border-color: #F9FAFB #F9FAFB #606060;
        height: 40px;
        left: 320px;
        top: 300px; 
    }
    .role{ 
        padding-left: 10px;
        position: absolute;
        font-style: normal;
        font-size: 15px;
        left: 220px;
        top: 400px;  
    }
    .OptionSelect{
        position: absolute;
        width: 450px;
        border-color: #F9FAFB #F9FAFB #606060;
        height: 40px;
        left: 320px;
        top: 450px; 
    
    }
    .buttonClass{
        position: absolute;
        display: flex;
        flex-direction: row;
        justify-content: center;
        margin-left: 300px;
        align-items: center;
        padding: 10px 60px;
        position: absolute;
        bottom: 180px;
        gap: 10px;
        right: 800px;
        background: #395B64;
        }
        .buttonText{
          color: white;
          size:100px
      }

  }
