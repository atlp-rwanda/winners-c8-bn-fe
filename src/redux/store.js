// this file will configure the redux's store
import { createStore } from "redux";
import rootReducer from "./reducers";

let authToken = ""; 
if(window.localStorage.getItem("auth-token")) {
  authToken = window.localStorage.getItem("auth-token");
}

const initialState = {
    auth: {token: authToken},
    navbar: {currentPage: "HOME"},
};
const store = createStore(
    rootReducer,
    initialState // sets initial state
  );

export default store;
