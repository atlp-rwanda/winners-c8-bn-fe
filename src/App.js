import React from 'react';
import Home from './components/Home';
import { Route, Routes } from 'react-router-dom';
import UserRegisterForm from './components/RegisterForm';
import RecoveryForm from './components/recoveryForm';
import ResetForm from './components/resetForm';
import LoginForm from './components/LoginForm';
import Userprofile from './components/UserProfile/UserProfile';
import Request from './pages/Request';
import Comments from './pages/Comments';
import Dashboard from './components/Dashboard';
import NotFound from './components/NotFound';
import UserRoleDash from "./components/AssignRole/UserRoleDash"
import { ToastContainer } from 'react-toastify';
import DashboardLayout from './components/Layouts/Dashboard';
import HomePageLayout from './components/Layouts/Home';

const App = () => {
  return (
    <div className="App">
      <ToastContainer />
      <div className="content">
        <Routes>
          <Route path="/" element={<HomePageLayout />}>
            <Route exact path="login" element={<LoginForm />} />
            <Route index element={<Home />} />
            <Route exact path="register" element={<UserRegisterForm />} />
            <Route exact path="recover" element={<RecoveryForm />} />
            <Route exact path="reset" element={<ResetForm />} />
          </Route>
          <Route path="/dashboard" element={<DashboardLayout />}>
            <Route exact index element={<Dashboard />} />
            <Route path="trips" element={<Request />} />
            <Route path="trips/trip" element={<Comments />} />
          <Route exact path='assignRole' element={<UserRoleDash />} />
            <Route exact path="userprofile" element={<Userprofile />} />
            <Route path="*" element={<NotFound />} />
          </Route>
          <Route exact path="*" element={<NotFound />} />
        </Routes>
      </div>
      {/* <SocialAuth /> */}
    </div>
  );
};

export default App;
