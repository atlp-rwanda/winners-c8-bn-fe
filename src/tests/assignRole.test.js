import React from "react";
import UserRoleDash from "../components/AssignRole/UserRoleDash";
import renderer from 'react-test-renderer'
import { cleanup,render, act } from "@testing-library/react";
import '@testing-library/jest-dom';
import store from '../redux/store';
import App from '../App';
import { MemoryRouter} from 'react-router-dom';
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureMockStore from "redux-mock-store";
const middleware = [thunk];
const mockStore = configureMockStore(middleware);

const roleInitialState = {
    UserRoleDash: {
      requestSent: false,
      responseData: {
        isSuccess: undefined,
        message: undefined,
      },
    },
  };



  afterEach(() => {
    cleanup;
  });


 
test("<UserRoleDash /> matches snapshot", ()=>{   
  const component = renderer
		.create(
			<Provider store={store}>
				<MemoryRouter>
          <UserRoleDash />
				</MemoryRouter>
			</Provider>,
		)
		.toJSON();
    expect(component).toMatchSnapshot();
})