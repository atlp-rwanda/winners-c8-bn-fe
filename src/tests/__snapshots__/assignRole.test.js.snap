// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserRoleDash /> matches snapshot 1`] = `
<div
  className="Content"
  data-testid="assign-role"
>
  <div
    className="sidebar"
  >
    <div
      className="top"
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <img
          alt="logo barefoot"
          className="logo"
          src=""
          width={100}
        />
      </a>
    </div>
    <hr />
    <div
      className="center"
    >
      <ul>
        <p
          className="title"
        >
          MAIN
        </p>
        <a
          href="/dashboard"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          <li>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="DashboardIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
              />
            </svg>
            <span>
              Dashboard
            </span>
          </li>
        </a>
        <p
          className="title"
        >
          PAGES
        </p>
        <a
          href="/users"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          <li>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="PersonOutlineIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
              />
            </svg>
            <span>
              Create locat&accom
            </span>
          </li>
        </a>
        <a
          href="/assignRole"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          <li>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="StoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4v2h16V4zm1 10v-2l-1-5H4l-1 5v2h1v6h10v-6h4v6h2v-6h1zm-9 4H6v-4h6v4z"
              />
            </svg>
            <span>
              Assign Roles
            </span>
          </li>
        </a>
        <li>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
            data-testid="CreditCardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zm0 14H4v-6h16v6zm0-10H4V6h16v2z"
            />
          </svg>
          <span>
            Assign Managers
          </span>
        </li>
        <p
          className="title"
        />
        <a
          href="/dashboard/trips"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          <li>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="InsertChartIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z"
              />
            </svg>
            <span>
              Trips
            </span>
          </li>
        </a>
        <li>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
            data-testid="NotificationsNoneIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.9 2 2 2zm6-6v-5c0-3.07-1.63-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.64 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2zm-2 1H8v-6c0-2.48 1.51-4.5 4-4.5s4 2.02 4 4.5v6z"
            />
          </svg>
          <span>
            Widgets
          </span>
        </li>
        <li>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
            data-testid="NotificationsNoneIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.9 2 2 2zm6-6v-5c0-3.07-1.63-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.64 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2zm-2 1H8v-6c0-2.48 1.51-4.5 4-4.5s4 2.02 4 4.5v6z"
            />
          </svg>
          <span>
            Maps
          </span>
        </li>
        <p
          className="title"
        >
          SOCIAL
        </p>
        <li>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
            data-testid="SettingsSystemDaydreamOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 17H9c-2.21 0-4-1.79-4-4 0-1.93 1.36-3.56 3.22-3.92C9.04 7.8 10.47 7 12 7c1.95 0 3.66 1.28 4.26 3.09 1.58.36 2.74 1.75 2.74 3.41 0 1.93-1.57 3.5-3.5 3.5zm-6.76-5.98C7.74 11.15 7 11.99 7 13c0 1.1.9 2 2 2h6.5c.83 0 1.5-.67 1.5-1.5s-.67-1.5-1.5-1.5h-.87l-.17-.86C14.29 9.92 13.23 9 12 9c-.96 0-1.84.57-2.26 1.45l-.27.57h-.73zM21 3H3c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16.01H3V4.99h18v14.02z"
            />
          </svg>
          <span>
            Notifications
          </span>
        </li>
        <li>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
            data-testid="PsychologyOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m15.82 7.22-1 .4c-.21-.16-.43-.29-.67-.39L14 6.17c-.02-.1-.1-.17-.2-.17h-1.6c-.1 0-.18.07-.19.17l-.15 1.06c-.24.1-.47.23-.67.39l-1-.4c-.09-.03-.2 0-.24.09l-.8 1.38c-.05.09-.03.2.05.26l.85.66c-.03.12-.05.26-.05.39s.01.26.03.39l-.84.66c-.08.06-.1.17-.05.25l.8 1.39c.05.09.15.12.25.09l.99-.4c.21.16.43.29.68.39l.14 1.06c.02.1.1.17.2.17h1.6c.1 0 .18-.07.2-.17l.15-1.06c.24-.1.47-.23.67-.39l.99.4c.09.04.2 0 .24-.09l.8-1.39c.05-.09.03-.19-.05-.25l-.83-.66c.02-.13.03-.26.03-.39 0-.14-.01-.27-.03-.39l.85-.66c.08-.06.1-.17.05-.26l-.8-1.38c-.05-.09-.16-.12-.25-.09zM13 11.43c-.79 0-1.43-.64-1.43-1.43s.64-1.43 1.43-1.43 1.43.64 1.43 1.43-.64 1.43-1.43 1.43z"
            />
            <path
              d="M19.94 9.06c-.43-3.27-3.23-5.86-6.53-6.05C13.27 3 13.14 3 13 3 9.47 3 6.57 5.61 6.08 9l-1.93 3.48c-.41.66.07 1.52.85 1.52h1v2c0 1.1.9 2 2 2h1v3h7v-4.68c2.62-1.25 4.35-4.08 3.94-7.26zm-5.05 5.57-.89.42V19h-3v-3H8v-4H6.7l1.33-2.33C8.21 7.06 10.35 5 13 5c2.76 0 5 2.24 5 5 0 2.09-1.29 3.88-3.11 4.63z"
            />
          </svg>
          <span>
            Chatting
          </span>
        </li>
        <li>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
            data-testid="SettingsApplicationsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm7-7H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-1.75 9c0 .23-.02.46-.05.68l1.48 1.16c.13.11.17.3.08.45l-1.4 2.42c-.09.15-.27.21-.43.15l-1.74-.7c-.36.28-.76.51-1.18.69l-.26 1.85c-.03.17-.18.3-.35.3h-2.8c-.17 0-.32-.13-.35-.29l-.26-1.85c-.43-.18-.82-.41-1.18-.69l-1.74.7c-.16.06-.34 0-.43-.15l-1.4-2.42c-.09-.15-.05-.34.08-.45l1.48-1.16c-.03-.23-.05-.46-.05-.69 0-.23.02-.46.05-.68l-1.48-1.16c-.13-.11-.17-.3-.08-.45l1.4-2.42c.09-.15.27-.21.43-.15l1.74.7c.36-.28.76-.51 1.18-.69l.26-1.85c.03-.17.18-.3.35-.3h2.8c.17 0 .32.13.35.29l.26 1.85c.43.18.82.41 1.18.69l1.74-.7c.16-.06.34 0 .43.15l1.4 2.42c.09.15.05.34-.08.45l-1.48 1.16c.03.23.05.46.05.69z"
            />
          </svg>
          <span>
            Settings
          </span>
        </li>
        <p
          className="title"
        >
          GENERAL
        </p>
        <a
          href="/dashboard/userprofile"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          <li>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="AccountCircleOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33C4.62 15.49 4 13.82 4 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83zM12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
              />
            </svg>
            <span>
              Profile
            </span>
          </li>
        </a>
        <li>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
            data-testid="ExitToAppIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10.09 15.59 11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <span>
            Logout
          </span>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="ContentContainer"
  >
    <div
      className="navbar"
    >
      <div
        className="wrapper"
      >
        <div
          className="search"
        >
          <input
            placeholder="Search..."
            type="text"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <div
          className="items"
        >
          <div
            className="item"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="LanguageOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2s.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2s.07-1.35.16-2h4.68c.09.65.16 1.32.16 2s-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2s-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
              />
            </svg>
          </div>
          <div
            className="item"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DarkModeOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9.37 5.51c-.18.64-.27 1.31-.27 1.99 0 4.08 3.32 7.4 7.4 7.4.68 0 1.35-.09 1.99-.27C17.45 17.19 14.93 19 12 19c-3.86 0-7-3.14-7-7 0-2.93 1.81-5.45 4.37-6.49zM12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9 9-4.03 9-9c0-.46-.04-.92-.1-1.36-.98 1.37-2.58 2.26-4.4 2.26-2.98 0-5.4-2.42-5.4-5.4 0-1.81.89-3.42 2.26-4.4-.44-.06-.9-.1-1.36-.1z"
              />
            </svg>
          </div>
          <div
            className="item"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="EmailOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
              />
            </svg>
            <div
              className="counter_chat"
            >
              2
            </div>
          </div>
          <div
            className="item"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="NotificationsNoneOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.9 2 2 2zm6-6v-5c0-3.07-1.63-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.64 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2zm-2 1H8v-6c0-2.48 1.51-4.5 4-4.5s4 2.02 4 4.5v6z"
              />
            </svg>
            <div
              className="counter"
            />
          </div>
          <div
            className="item"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="ListOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 13h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7zm-4 6h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7z"
              />
            </svg>
          </div>
          <div
            className="item"
          >
            <img
              alt=""
              className="avatar"
              src="https://images.pexels.com/photos/1680172/pexels-photo-1680172.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2"
            />
          </div>
          <div
            className="item"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="SettingsOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.43 12.98c.04-.32.07-.64.07-.98 0-.34-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.09-.16-.26-.25-.44-.25-.06 0-.12.01-.17.03l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.06-.02-.12-.03-.18-.03-.17 0-.34.09-.43.25l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98 0 .33.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.09.16.26.25.44.25.06 0 .12-.01.17-.03l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.06.02.12.03.18.03.17 0 .34-.09.43-.25l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zm-1.98-1.71c.04.31.05.52.05.73 0 .21-.02.43-.05.73l-.14 1.13.89.7 1.08.84-.7 1.21-1.27-.51-1.04-.42-.9.68c-.43.32-.84.56-1.25.73l-1.06.43-.16 1.13-.2 1.35h-1.4l-.19-1.35-.16-1.13-1.06-.43c-.43-.18-.83-.41-1.23-.71l-.91-.7-1.06.43-1.27.51-.7-1.21 1.08-.84.89-.7-.14-1.13c-.03-.31-.05-.54-.05-.74s.02-.43.05-.73l.14-1.13-.89-.7-1.08-.84.7-1.21 1.27.51 1.04.42.9-.68c.43-.32.84-.56 1.25-.73l1.06-.43.16-1.13.2-1.35h1.39l.19 1.35.16 1.13 1.06.43c.43.18.83.41 1.23.71l.91.7 1.06-.43 1.27-.51.7 1.21-1.07.85-.89.7.14 1.13zM12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 6c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      className="titleDashboard"
    >
      <div
        className="text"
      >
        <h4>
          Dashboard&gt;&gt;
        </h4>
        <h6>
          Assign Role
        </h6>
      </div>
      <h4>
        Dashboard&gt;&gt;
      </h4>
    </div>
    <div
      className="contentBlock"
    >
      <p>
        Assign Role
      </p>
      <span
        className="MuiSkeleton-root MuiSkeleton-rectangle MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
        style={
          Object {
            "height": 800,
            "width": 1100,
          }
        }
      />
    </div>
  </div>
</div>
`;
